hanadb_exporter/utils.py

```diff
@@ -0,0 +1,52 @@
+"""
+Generic methods
+
+:author: xarbulu
+:organization: SUSE Linux GmbH
+:contact: xarbulu@suse.de
+
+:since: 2019-07-04
+"""
+
+from distutils import version
+
+# TODO: this method could go in shaptools itself, providing the query return formatted if
+# it is requested (returning a list of dictionaries like this method)
+def format_query_result(query_result):
+    """
+    Format query results to match column names with their values for each row
+    Returns: list containing a dictionaries (column_name, value)
+
+    Args:
+        query_result (obj): QueryResult object
+    """
+    formatted_query_result = []
+    query_columns = [meta[0] for meta in query_result.metadata]
+    for record in query_result.records:
+        record_data = {}
+        for index, record_item in enumerate(record):
+            record_data[query_columns[index]] = record_item
+
+        formatted_query_result.append(record_data)
+    return formatted_query_result
+
+
+def check_hana_range(hana_version, availability_range):
+    """
+    Check if the current hana version is inside the available range
+
+    Args:
+        hana_version (str): Current hana version
+        availability_range (list): List with one or two elements definining the
+            available hana versions
+
+    Returns:
+        bool: True if the current hana version is inside the availability range
+    """
+
+    if len(availability_range) == 1:
```

Comment:
`nit`. I would prefer in term of readabilty a dispatch on dictionary like this
```python
#! /usr/bin/python3

def handle_one():
  print("one")

def handle_two():
  print("two")

def handle_three():
  print("bro")


def handle_options(option):
  {'one': handle_one,
   2: handle_two,
   'three': handle_three}[option]()

handle_options(2)```

It would be imho easier when we want to handle other version cases maybe?

Reply:
I really hope that this method won't have more options... The range shouldn't never have more than two options... If we needed to add more, I agree that changing to your approach could a good idea. By now, I think that the current code is good enough (an quite self-explicative)